Index: scripts/check-files
===================================================================
--- scripts/check-files
+++ scripts/check-files	2007-02-28 18:39:01.486363480 +0100
@@ -13,12 +13,14 @@
 	exit 1
 fi
 
+RPM_BUILD_ROOT=`echo $RPM_BUILD_ROOT | sed 's://*:/:g'`
 [ "$TMPDIR" ] || TMPDIR=/tmp
 FILES_DISK=`mktemp $TMPDIR/rpmXXXXXX`
 FILES_RPM=`mktemp $TMPDIR/rpmXXXXXX`
 
-find $RPM_BUILD_ROOT -type f -o -type l | LC_ALL=C sort > $FILES_DISK
-LC_ALL=C sort -u > $FILES_RPM
+
+find $RPM_BUILD_ROOT -type f -o -type l | sed 's://*:/:g' | LC_ALL=C sort > $FILES_DISK
+sed 's://*:/:g' | LC_ALL=C sort -u > $FILES_RPM
 
 diff "$FILES_DISK" "$FILES_RPM" | grep "^< " | cut -c3- | sed -e "s#^$RPM_BUILD_ROOT#   #g"
 
