--- rpm-4.4.6/popt/popt.c.pix	2006-02-12 21:57:20.000000000 +0100
+++ rpm-4.4.6/popt/popt.c	2006-11-22 18:58:31.000000000 +0100
@@ -463,7 +463,9 @@
 
     rc = execvp(argv[0], (char *const *)argv);
 
-    return POPT_ERROR_ERRNO;
+    /* FIXME: hardcoding error message and exit status */
+    fprintf(stderr, _("Could not exec %s: %s\n"), argv[0], strerror(errno));
+    exit(EXIT_FAILURE);
 }
 /*@=bounds =boundswrite @*/
 
