diff -up rpm-4.4.2.2/rpmdb/sqlite.c.pix rpm-4.4.2.2/rpmdb/sqlite.c
--- rpm-4.4.2.2/rpmdb/sqlite.c.pix	2007-10-23 12:24:41.000000000 +0200
+++ rpm-4.4.2.2/rpmdb/sqlite.c	2007-10-23 12:30:41.000000000 +0200
@@ -694,13 +694,15 @@ static int sql_initDB(dbiIndex dbi)
 	}
 if (_debug)
 fprintf(stderr, "\t%s(%d) type(%d) keytype %s\n", tagName(dbi->dbi_rpmtag), dbi->dbi_rpmtag, tagType(dbi->dbi_rpmtag), keytype);
-	sprintf(cmd, "CREATE TABLE '%s' (key %s, value %s)",
+	sprintf(cmd, "CREATE %sTABLE '%s' (key %s, value %s)",
+			dbi->dbi_rpmtag == RPMDBI_DEPENDS ? "TEMPORARY " : " ",
 			dbi->dbi_subfile, keytype, valtype);
 	rc = sqlite3_exec(sqldb->db, cmd, NULL, NULL, (char **)&scp->pzErrmsg);
 	if (rc)
 	    goto exit;
 
-	sprintf(cmd, "CREATE TABLE 'db_info' (endian TEXT)");
+	sprintf(cmd, "CREATE %sTABLE 'db_info' (endian TEXT)",
+			dbi->dbi_rpmtag == RPMDBI_DEPENDS ? "TEMPORARY " : " ");
 	rc = sqlite3_exec(sqldb->db, cmd, NULL, NULL, (char **)&scp->pzErrmsg);
 	if (rc)
 	    goto exit;
