diff -up rpm-4.4.2.2/configure.ac.pix rpm-4.4.2.2/configure.ac
--- rpm-4.4.2.2/configure.ac.pix	2007-10-23 15:02:32.000000000 +0200
+++ rpm-4.4.2.2/configure.ac	2007-10-23 15:19:36.000000000 +0200
@@ -1269,7 +1269,7 @@ AC_DEFINE_UNQUOTED(SYSCONFIGDIR, "$SYSCO
 	[Full path to rpm system configuration directory (usually /etc/rpm44)])
 AC_SUBST(SYSCONFIGDIR)
 
-MACROFILES="${RPMCONFIGDIR}/macros:${RPMCONFIGDIR}/%{_target_platform}/macros:${SYSCONFIGDIR}/macros:${SYSCONFIGDIR}/macros.specspo:${SYSCONFIGDIR}/macros.db1:${SYSCONFIGDIR}/%{_target_platform}/macros:~/.rpmmacros"
+MACROFILES="${RPMCONFIGDIR}/macros:${RPMCONFIGDIR}/%{_target_platform}/macros:${RPMCONFIGDIR}/%{_host_vendor}/macros:${RPMCONFIGDIR}/%{_host_vendor}/%{_target_platform}/macros:${SYSCONFIGDIR}/macros.d/*.macros:${SYSCONFIGDIR}/macros:${SYSCONFIGDIR}/macros.specspo:${SYSCONFIGDIR}/macros.db1:${SYSCONFIGDIR}/%{_target_platform}/macros:~/.rpmmacros"
 AC_DEFINE_UNQUOTED(MACROFILES, "$MACROFILES",
 	[Colon separated paths of macro files to read.])
 AC_SUBST(MACROFILES)
